src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/DiscardParameterHighlighter.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editor.Implementation.Highlighting;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.Highlighting.KeywordHighlighters
+{
+    [ExportHighlighter(LanguageNames.CSharp)]
+    internal class DiscardParameterHighlighter : AbstractKeywordHighlighter
+    {
+        [ImportingConstructor]
+        public DiscardParameterHighlighter()
+        {
+        }
+
+        protected override bool IsHighlightableNode(SyntaxNode node)
+        {
+            if (!node.IsKind(CodeAnalysis.CSharp.SyntaxKind.Argument) || !(node is ArgumentSyntax syntax))
+            {
+                return false;
+            }
+
+            if (!syntax.RefOrOutKeyword.IsKind(SyntaxKind.OutKeyword))
+            {
+                return false;
+            }
+
+            if (syntax.Expression.IsKind(SyntaxKind.IdentifierName) && syntax.Expression is IdentifierNameSyntax name)
+            {
+                return name.Identifier.Text == "_";
+            }
+
+            if (syntax.Expression.IsKind(SyntaxKind.DeclarationExpression) && syntax.Expression is DeclarationExpressionSyntax declaration)
+            {
+                return declaration.Designation.IsKind(SyntaxKind.DiscardDesignation);
+            }
+
+            return false;
+        }
+
+        protected override IEnumerable<TextSpan> GetHighlightsForNode(SyntaxNode node, CancellationToken cancellationToken)
+        {
+            var argument = (ArgumentSyntax)node;
+
+            if (argument.Expression.IsKind(SyntaxKind.IdentifierName))
```

Comment:
can do an `is` pattern check so you don't need to check both the kind and then also do a cast.